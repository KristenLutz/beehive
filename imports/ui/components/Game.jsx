// @flow

// Framework
import React from "react";

// Components
import { Container, Row, Col, Button } from "reactstrap";

/**
 * The main game element.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const Game = ({
  bees = [],
  createBee
}: {
  bees: Array<Bee>,
  createBee: () => void
}): StatelessComponent =>
  <Container className="game" fluid>
    <Row noGutters>
      <Col>
        <p>Game</p>
      </Col>
    </Row>
    <Row noGutters>
      <Col>
        <Container fluid>
          <Row noGutters>
            <Col>
              <Button onClick={createBee}>Create New Bee</Button>
            </Col>
          </Row>
          <Row noGutters>
            <Col>
              <p>
                <strong>Bees:</strong> {bees.length}
              </p>
            </Col>
          </Row>
        </Container>
      </Col>
    </Row>
  </Container>;

export default Game;
