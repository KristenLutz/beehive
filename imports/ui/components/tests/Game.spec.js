import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";

import Game from "../Game";

configure({ adapter: new Adapter() });

describe("<Game />", () => {
  const defaultProps = {
    bees: [],
    removeBee: jest.fn()
  };

  describe("Game SnapShot", () => {
    it("renders Game correctly", () => {
      const tree = ReactTestRenderer.create(
        <Game {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
