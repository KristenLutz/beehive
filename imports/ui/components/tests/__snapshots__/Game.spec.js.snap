// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> Game SnapShot renders Game correctly 1`] = `
<div
  className="game container-fluid"
>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <p>
        Game
      </p>
    </div>
  </div>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <p>
        <strong>
          Total Bees:
        </strong>
         
        0
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="card-columns"
    />
  </div>
</div>
`;
